kernel/src/engine/parquet_row_group_skipping.rs

```diff
@@ -179,8 +179,15 @@ impl<'a> ParquetStatsSkippingFilter for RowGroupFilter<'a> {
     // Parquet nullcount stats always have the same type (u64), so we can directly return the value
     // instead of wrapping it in a Scalar. We can safely cast it from u64 to i64, because the
     // nullcount can never be larger than the rowcount, and the parquet rowcount stat is i64.
+    //
+    // NOTE: Stats for any given column are optional, which may produce a NULL nullcount. But if
+    // the column itself is missing, then we know all values are implied to be NULL.
     fn get_nullcount_stat_value(&self, col: &ColumnPath) -> Option<i64> {
-        Some(self.get_stats(col)?.null_count_opt()? as i64)
+        let nullcount = match self.get_stats(col) {
+            Some(s) => s.null_count_opt()? as i64,
+            None => self.get_rowcount_stat_value(),
```

Comment:
This is a new find, exposed on accident by me hacking two more parts into the checkpoint so we could test transaction app id filtering (the "checkpoint" schema was truncated, which prevented the P&M query from skipping those parts)

Reply:
If the [`statistics()`](https://arrow.apache.org/rust/parquet/file/metadata/struct.ColumnChunkMetaData.html#method.statistics) method on `ColumnChunkMetadata` returns `None`, that just means that there are no stats for that column, but doesn't necessarily imply that _all_ values are `null` does it?

Reply:
Oh, good catch. I didn't put the check deep enough. There are three levels of `None` here:
* The column chunk doesn't even exist (infer nullcount = rowcount)
* The column chunk doesn't have stats (should _not_ infer anything clever)
* The stats object lacks a particular stat

To make things even more "fun", we have the following warning in [Statistics::null_count_opt](https://arrow.apache.org/rust/parquet/file/statistics/enum.Statistics.html#method.null_count_opt) 🤦:
> this API returns Some(0) even if the null count was not present in the statistics

So I have two problems to work around now.

Reply:
Both fixed.